app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -63,6 +63,18 @@ class PatientRepository @Inject constructor(
     }
   }
 
+  @WorkerThread
+  fun search2(criteria: PatientSearchCriteria, facilityId: UUID): List<PatientSearchResult> {
+    return when (criteria) {
+      is Name -> searchByPhoneNumber2(criteria.patientName, facilityId) //Change to searchByName2
+      is PhoneNumber -> searchByPhoneNumber2(criteria.phoneNumber, facilityId)
+    }
+  }
```

Comment:
This is not what I suggested 😅 . What I meant was something like this.

```kt
  @WorkerThread
  fun search2(criteria: PatientSearchCriteria, facilityId: UUID): List<PatientSearchResult> {
    return  searchByPhoneNumber2(criteria.phoneNumber, facilityId)
  }
```

Here at this point, we are only returning phone number results. Once we are adding the search by name functionality then we introduce the conditional to switch between those two. So we are following the [YAGNI principle](https://martinfowler.com/bliki/Yagni.html).

Reply:
That's not gonna work as criteria can either be PhoneNumber of Name. If it's PhoneNumber then we get the number string. I can call the old searchByName() method here.

Reply:
Ah, you can cast the criteria. So it will be like
```kt
val phoneNumber = (criteria as PhoneNumber).phoneNumber
searchByPhoneNumber2(phoneNumber, facilityId)
```

Reply:
Since in this commit we are only searching for phone number this should be fine. Once we add the search by name in next commit it should be fine, we can add that conditional there.